### ISSUE TITLE
Load the tabs on-demand when a container is shown?

### ISSUE THREAD

**Issue by anfelor** [mozilla/testpilot-containers#348]
When Firefox starts not all tabs are loaded: The websites are requested when needed.

On the other hand, when a container is shown, all tabs in this container will be loaded.
This makes using containers for tab management (#336) hard,
as it takes longer to switch between containers with many open tabs.

**Comment by johngruen** [comment]
Probably the conservative course here would be to treat re-shown tabs the way we do on session restore. (blurred tabs inactive, pinned and focused tabs loaded)

**Comment by smichel17** [comment]
> Probably the conservative course here would be to treat re-shown tabs the way we do on session restore. (blurred tabs inactive, pinned and focused tabs loaded)

What other course is available?

**Comment by anfelor** [comment]
Has there been any progress on this issue?

**Comment by groovecoder** [comment]
Sorry, no. There are other issues with more votes. We include issue #'s in our commit messages, so if we work on this one, it should automatically update this issue.

**Comment by falolaf** [comment]
This would be a most welcome addition indeed!

**Comment by jonathanKingston** [comment]
@SoftVision-PaulOiegas said in #290

Ok, but consider this scenario. You have lots of hidden pages in a specific container (eg: 50) and you want to open a new tab of that container. This means that all the hidden tabs are opened and loaded for just one new tab.
Wouldn't this affect A LOT the user experience and browser performance since it will take a lot of time to load each and every hidden page ? Because the add-on also loads the page content, not only displays them as the "Remember tabs and windows from last session" option does.

Also, probably the users will use the option to hide personal tabs because he doesn't want others to see those tabs. Forcing all of them to be opened when user opens a new tab will automatically expose those tabs.

In my opinion, none of this is right, but hey, maybe it's just me...

**Comment by grahamperrin** [comment]
I have not observed this issue. Was the enhancement request fulfilled?

**Comment by falolaf** [comment]
@grahamperrin It has not been fullfilled in my case at least. All tabs are loaded when a container is unhidden.

**Comment by grahamperrin** [comment]
Is this still an issue? 

I tested with [Conex](https://addons.mozilla.org/addon/conex/) set to: 

☑ only show tabs of the current container

– with multiple tabs in each container, then quit, then started Firefox `88`. 

Switching to a different container did not load all tabs in the container.

**Comment by eMBee** [comment]
i just tested with https://github.com/mozilla/multi-account-containers

1. create a container.
2. add some tabs and load sites in them.
3. hide the container.
4. show the container
5. tabs in the container are shown, but sites are not loaded until each tabs is opened

**Comment by grahamperrin** [comment]
Thanks for confirming. 

Incidentally: 

> 3. hide the container.

– is this achievable with the Multi-Account Containers extension alone? (Am I missing something?)

**Comment by eMBee** [comment]
> > 3. hide the container.
> – is this achievable with the Multi-Account Containers extension alone? (Am I missing something?)

yes, this was always part of the features as far as i can remember.

**Comment by grahamperrin** [comment]
I can't find the feature …
